lib/connection.ts

```diff
@@ -264,17 +294,20 @@ export class Connection extends Entity {
 
   /**
    * Creates a new amqp connection.
+   * @param options A set of options including a signal used to cancel the operation.
    * @return {Promise<Connection>} Promise<Connection>
    * - **Resolves** the promise with the Connection object when rhea emits the "connection_open" event.
    * - **Rejects** the promise with an AmqpError when rhea emits the "connection_close" event
-   * while trying to establish an amqp connection.
+   * while trying to establish an amqp connection or with an AbortError if the operation was cancelled.
    */
-  open(): Promise<Connection> {
+  open(options?: ConnectionOpenOptions): Promise<Connection> {
     return new Promise((resolve, reject) => {
       if (!this.isOpen()) {
 
         let onOpen: Func<RheaEventContext, void>;
         let onClose: Func<RheaEventContext, void>;
+        let onAbort: Func<void, void>;
+        const abortSignal = options?.abortSignal;
```

Comment:
I suppose this must work or else CI wouldn't run but the package.json says we're using TS 3.5.

Reply:
That's because 3.5.1 is the minimum version :smile: But that does remind me that before a new version is published, we should evaluate if down-leveled types are needed.

Reply:
Ah, right! :)

It's a dev dependency anyways - so should we bump it up?

Reply:
Removed optional chaining in https://github.com/amqp/rhea-promise/pull/66/commits/c12aeed195531b48ce9a06a5d83f5c5fd88fba5b to avoid rocking the boat regarding TS versions